src/lib.rs

```diff
@@ -130,6 +130,9 @@ impl<'a> DoubleArrayTrieBuilder<'a> {
 
     // pub fn build(mut self, keys: &[&str], values: &[usize]) -> DoubleArrayTrie {
     pub fn build(mut self, keys: &'a [&str]) -> DoubleArrayTrie {
+        //using the unicode scalar len is correct since that's our DARTS unit here
+        let longest_word_len = keys.iter().map(|s| s.len()).max().unwrap_or(0);
```

Comment:
``common_prefix_search`` returns ``char`` code points. ``.chars().count()`` may be a better choice.